pandas/core/tools/datetimes.py

```diff
@@ -390,11 +393,17 @@ def _convert_listlike_datetimes(
             format = None
 
     if format is not None:
-        res = _to_datetime_with_format(
-            arg, orig_arg, name, tz, format, exact, errors, infer_datetime_format
-        )
-        if res is not None:
-            return res
+        try:
+            res = _to_datetime_with_format(
+                arg, orig_arg, name, tz, format, exact, errors, infer_datetime_format
+            )
+            if res is not None:
+                return res
+        except ValueError:
+            # GH#42957: ValueError: time data '<NA>'
+            # does not match format '%Y%m%d%H%M%S'
+            if isna(arg):
+                format = None
```

Comment:
@debnathshoham 

It doesn't fix BUG. I'm sorry for bad desription, but i have bug with this code too:

```python
import numpy as np
import pandas as pd

a = pd.DataFrame({'a':[pd.NA, "20210202202020"]})  
b = pd.DataFrame({'b': [np.nan, "20210202202020"]})
print(pd.to_datetime(b['b'], format="%Y%m%d%H%M%S"))      # -> Work
# 0                   NaT
# 1   2021-02-02 20:20:20
# Name: b, dtype: datetime64[ns]
print(pd.to_datetime(a['a'], format="%Y%m%d%H%M%S"))       # -> Error
#ValueError: time data '<NA>' does not match format '%Y%m%d%H%M%S' (match)
```

So datetime formatter can't work with `pd.NA` values but for `np.nan` everything is good.

Reply:
I think that bug is [here](https://github.com/pandas-dev/pandas/blob/a94fd287f35d0a50a5e0c1455a7bff276eb35edf/pandas/_libs/tslibs/nattype.pyx#L40) 

`pd.NA` is unknown `nantype_string`

Reply:
Hey @levkovalenko, I think this should be work now.